weblab_cg/_generator_chaste_cg.py

```diff
@@ -0,0 +1,627 @@
+
+    def _add_printers(self):
+        """ Initialises Printers for outputting chaste code. """
+        self._var_chaste_interface_printer =\
+            cg.ChastePrinter(lambda symbol: 'var_chaste_interface_' + str(symbol).replace('$', '__'),
+                             lambda deriv: 'd_dt_chaste_interface_' + (str(deriv.expr).replace('$', '__')
+                             if isinstance(deriv, sp.Derivative) else str(deriv).replace('$', '__')))
+
+        # Printer for printing chaste regular variable assignments (var_ prefix)
+        self._printer = \
+            cg.ChastePrinter(lambda symbol, state_vars=self._state_vars:
+                             ('var_chaste_interface_' if symbol in state_vars else 'var')
+                             + str(symbol).replace('$', '__'),
+                             lambda deriv: 'd_dt_chaste_interface_' +
+                                           (str(deriv.expr).replace('$', '__')
+                                            if isinstance(deriv, sp.Derivative) else str(deriv).replace('$', '__')))
+
+        # Printer for printing chaste regular variable assignments without the var_ prefix
+        self._name_printer = cg.ChastePrinter(lambda symbol: str(symbol)[1:].replace('$', '__'))
+
+    def _add_units(self):
+        """ Add needed units to the model to allow converting time, voltage and calcium in specific units."""
+        for unit_name in self._UNIT_DEFINITIONS:
+            try:
+                self._model.units.add_custom_unit(unit_name, self._UNIT_DEFINITIONS[unit_name])
+            except AssertionError:
+                pass  # Unit already exists, but that is not a problem
+
+    def _get_membrane_voltage_var(self):
+        """ Find the membrane_voltage variable"""
+        return self._model.get_symbol_by_ontology_term(self._OXMETA, "membrane_voltage")
+
+    def _get_cytosolic_calcium_concentration_var(self):
+        """ Find the cytosolic_calcium_concentration variable if it exists"""
+        try:
+            return self._model.get_symbol_by_ontology_term(self._OXMETA, "cytosolic_calcium_concentration")
+        except KeyError:
+            self._logger.debug(self._model.name + ' has no cytosolic_calcium_concentration')
+            return None
+
+    def _get_use_get_intracellular_calcium_concentration(self):
+        """ Get wyether or not the model has cytosolic_calcium_concentration"""
+        try:
+            self._model.get_symbol_by_ontology_term(self._OXMETA, "cytosolic_calcium_concentration")
+            return True
+        except KeyError:
+            return False
+
+    def _get_state_variables(self):
+        """ Sort the state variables, in similar order to pycml to prevent breaking existing code"""
+        return sorted(self._model.get_state_symbols(),
+                      key=lambda state_var: self._state_var_key_order(state_var))
+
+    def _get_membrane_stimulus_current(self):
```

Comment:
These methods, `_get_physical_variable`, are all very useful and should probably go in `cellmlmanip` (maybe in a separate module like [I'm planning](https://github.com/MichaelClerx/myokit/issues/351) for Myokit?). Maybe without the default bit though, so just a get_x method that returns x or raise a KeyError or returns None or something